app/uk/gov/hmrc/customs/managesubscription/services/SubscriptionCompleteBusinessService.scala

```diff
@@ -79,12 +78,14 @@ class SubscriptionCompleteBusinessService @Inject() (
       _         <- emailService.sendFailureEmail(recipient.recipientDetails)
     } yield (): Unit
 
-  private def dataStoreEmailRequest(
-    recipient: RecipientDetailsWithEori
-  )(implicit hc: HeaderCarrier): String => Future[HttpResponse] = eori =>
-    dataStoreConnector.storeEmailAddress(
-      DataStoreRequest(eori, recipient.recipientDetails.recipientEmailAddress, recipient.emailVerificationTimestamp)
-    )
+  private def dataStoreEmailRequest(recipient: RecipientDetailsWithEori)(implicit hc: HeaderCarrier): Future[Unit] =
+    retrieveEori(recipient).flatMap {
+      case Some(eori) if recipient.recipientDetails.service == "CDS" =>
```

Comment:
This field exists in the model before your changes. But I'll double check this tomorrow

Reply:
I'm sure I added it.  Not for emails specifically, but a while ago because we thought it would be useful.
It basically contains the "service" part of the url - in our case "atar".

Reply:
ah, okay.

I tried to understand what is in that field based on the tests. I saw `ATaR` that's the reason I thought you somehow store this.

In that case what do you think about storing the `Service` class in that field?